framework/src/engine/consensus/certificate_generation/commit_pool.ts

```diff
@@ -308,6 +312,7 @@ export class CommitPool {
 				maxHeightCertified + 1,
 			);
 			nextHeight = Math.min(heightNextBFTParameters - 1, maxHeightPrecommitted);
+			nextHeight = Math.max(nextHeight, this._minCertifyHeight - 1);
```

Comment:
I don't see why we use -1 here. 

My understanding is that `minCertifyHeight` we want to satisfy that  `getBFTParameters(minCertifyHeight)` would have the bls keys registered. So, it should be a first block of a round. For example, if all validators registered BLS keys during initRounds, it could be the first block of hybrid phase. 

If that's correct, then in order to produce a certificate, you need to have height at least `minCertifyHeight`, so I think the -1 should be dropped.